### PR TITLE
Fix changes to nested imports not updating styles during development

### PR DESCRIPTION
**Pull request by ArrayKnight**
Fixes https://github.com/vanilla-extract-css/vanilla-extract/issues/1427

Tracked down an issue where .css.ts files that are imported by .css.ts files trigger updates, but changes aren't propagated to rendered component during development. By invalidating all deps of a .css.ts file, instead of just the root file, the expected behavior occurs.

### CODE REVIEW

**Comment by ArrayKnight** [comment]
@askoufis I can't officially request a review, but I'd greatly appreciate your attention on this to try and get this merged soon. I'd like to minimize the impact on my team as much as possible. Thank you!

**Comment by askoufis** [comment]
@ArrayKnight Thanks for the fix! This looks correct to me. However, I'd like to inline our `addWatchFiles` function to avoid iterating over `watchFiles` twice. I can either do this in a separate PR (and close this one) or instead I could push to your fork, if you grant me permissions. LMK which you'd prefer.

**Comment by ArrayKnight** [comment]
@askoufis I just pushed an update that hopefully aligns with your goal. If not, I've added you as a collaborator on my fork. Take any steps you'd prefer to get this dialed in the way you'd like it

File: .changeset/chilled-paws-shop.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@vanilla-extract/vite-plugin': patch
+---
+
+Correctly update rendered styles when changes are made to .css.ts files imported by .css.ts files
```
Comment (askoufis) [inline .changeset/chilled-paws-shop.md:5]:
```suggestion
Fixes a bug where changes to `.css.ts` dependencies of top-level `.css.ts` files would not generate new CSS
```

**Review by askoufis — APPROVED** [review]

**Comment by askoufis** [comment]
Made a few small changes, just to the changeset and a formatting change, but otherwise you pretty much read my mind with that refactor!

Have tested this both with `vite dev` and `vite build --watch` and both seem to work how I'd expect.

**Pull request merged**